scripts/motor_driver.py

```diff
@@ -1,52 +1,96 @@
+        # Initialize pigpio
+        self.pi = pigpio.pi(host)
+        if not self.pi.connected:
+            rospy.signal_shutdown("Pigpio not connected, run `sudo pigpiod` or specify host")
+            exit()
 
-class Driver(DualMotorDriver):
-    def __init__(self):
-        super().__init__(pin['PWM1'],pin['DIR1'],pin['PWM2'],pin['DIR2'],pin['FAULT'],pin['EN'])
+        # Initialize motors
+        self.motorA = Motor(pin_out['PWM1'], pin_out['DIR1'], self.pi)
+        self.motorB = Motor(pin_out['PWM2'], pin_out['DIR2'], self.pi)
+
+        # Initialize fault pin
+        self.pin_FAULT = pin_out['FAULT']
+        self.pi.set_pull_up_down(self.pin_FAULT, pigpio.PUD_UP)  # make sure FAULT is pulled up
+
+        # Initialize enable pin
+        self.pin_EN = pin_out['EN']
+        self.pi.write(self.pin_EN, 0)  # enable drivers by default
 
         # Subscriber
         self.sub_front_right = rospy.Subscriber(
-            '/wheel_A_ctrl/command', Float64, self.setSpeedM1RosWrapper)
+            '/motor_A_ctrl/command', Float64, self.set_speed_M1)
         self.sub_front_left = rospy.Subscriber(
-            '/wheel_B_ctrl/command', Float64, self.setSpeedM2RosWrapper)
-        
+            '/motor_B_ctrl/command', Float64, self.set_speed_M2)
+
         # Publishers
         self.pub_fault_warning = rospy.Publisher('/fault_warning', Bool, queue_size=10)
-        rospy.loginfo("driver start")
-        rospy.loginfo("driver pinout: " +str(pin_out))
-        
 
-    def setSpeedM1RosWrapper(self, Float64_msg):
+        rospy.loginfo("Driver start")
+        rospy.loginfo("Driver pinout: " + str(pin_out))
+
+    def set_speed_M1(self, Float64_msg):
         m1_speed = Float64_msg.data
-        self.motor1.setSpeed(m1_speed)
-        rospy.logdebug("motor 1 speed:" + str(m1_speed))       
-    
-    def setSpeedM2RosWrapper(self, Float64_msg):
+        self.motorA.set_speed(m1_speed)
+        rospy.logdebug("Motor 1 speed:" + str(m1_speed))
+
+    def set_speed_M2(self, Float64_msg):
         m2_speed = Float64_msg.data
-        self.motor2.setSpeed(m2_speed) 
+        self.motorB.set_speed(m2_speed)
         rospy.logdebug("motor 2 speed:" + str(m2_speed))
 
-    def getFaultRosWrapper(self):
-        Bool_msg = Bool
-        Bool_msg.data = self.getFault()
-        rospy.logwarn("driver fault detected")
+    def get_fault(self):
+        Bool_msg = Bool(self.getFault())
+
+        rospy.logwarn("Driver fault detected")
         self.pub_fault_warning(Bool_msg)
 
-        
+    def enable(self):
+        rospy.logwarn("Motors enabled")
+        self.pi.write(self.pin_EN, 0)
+
+    def disable(self):
+        rospy.logwarn("Motors disabled")
+        self.pi.write(self.pin_EN, 1)
 
 
 if __name__ == '__main__':
-    rospy.init_node("Driver", anonymous=True)
-    Driver()
+    pins = rospy.get_param(sys.argv[2])
+    DualMotorDriver(host=sys.argv[1], pin_out=pins)
```

Comment:
Here is missing to take into account when it is run with no arguments.

```suggestion
    if len(sys.argv) > 1:
        pins = rospy.get_param(sys.argv[2])
        DualMotorDriver(host=sys.argv[1], pin_out=pins)
    else: 
        DualMotorDriver()
```

Reply:
this would not always work. For example, when I call the node from a `roslaunch`

```xml
    <rosparam file="$(find pai)/config/driver_L_pinout.yaml" command="load" />
    <node   name="left_driver" 
            pkg="pai" output="screen"
            type="motor_driver.py"     
     >   
        <remap from="/wheel_A_ctrl/command" to="/front_left_wheel_ctrl/command"/>
        <remap from="/wheel_B_ctrl/command" to="/back_left_wheel_ctrl/command"/>    
    </node>
```

the python node will receive more than 2 arguments due to the two `remap`.  I think we must manage this kind of error, but I do not know how yet.

Reply:
yeah you are right it would not work fix in the case of calling the node from a launch file without arguments. But it allows it to run the python file by itself, which I find useful when debugging.

Reply:
lgtm but I would write with more than 2 params due to the first argument is the file name

```suggestion
    if len(sys.argv) > 2:
        pins = rospy.get_param(sys.argv[2])
        DualMotorDriver(host=sys.argv[1], pin_out=pins)
    else: 
        DualMotorDriver()
```